tck/src/main/java/org/eclipse/microprofile/reactive/messaging/tck/channel/BeanInjectedWithDifferentFlavorsOfTheSameChannel.java

```diff
@@ -35,32 +37,61 @@
 public class BeanInjectedWithDifferentFlavorsOfTheSameChannel {
 
 
+    @Outgoing("hello-1")
+    public Publisher<String> hello1() {
+        return Flowable.fromArray("h", "e", "l", "l", "o");
+    }
+
+    @Outgoing("hello-2")
+    public Publisher<String> hello2() {
+        return Flowable.fromArray("h", "e", "l", "l", "o");
+    }
+
+    @Outgoing("hello-3")
+    public Publisher<String> hello3() {
+        return Flowable.fromArray("h", "e", "l", "l", "o");
+    }
+
+    @Outgoing("hello-4")
+    public Publisher<String> hello4() {
+        return Flowable.fromArray("h", "e", "l", "l", "o");
+    }
+
+    @Outgoing("hello-5")
+    public Publisher<String> hello5() {
+        return Flowable.fromArray("h", "e", "l", "l", "o");
+    }
+
+    @Outgoing("hello-6")
+    public Publisher<String> hello6() {
+        return Flowable.fromArray("h", "e", "l", "l", "o");
+    }
+
     @Inject
-    @Channel("hello")
+    @Channel("hello-1")
     private Publisher<Message<String>> field1;
 
     @Inject
-    @Channel("hello")
+    @Channel("hello-2")
     private Publisher<Message> field2;
 
 
     @Inject
-    @Channel("hello")
+    @Channel("hello-3")
     private PublisherBuilder<Message> field3;
 
     @Inject
-    @Channel("hello")
+    @Channel("hello-4")
     private PublisherBuilder<Message<String>> field4;
 
     @Inject
-    @Channel("hello")
+    @Channel("hello-5")
     private PublisherBuilder<String> field5;
 
     @Inject
-    @Channel("hello")
+    @Channel("hello-6")
     private Publisher<String> field6;
 
-  
 
     public Map<String, String> consume() {
         Map<String, String> map = new LinkedHashMap<>();
```

Comment:
I don't think this method makes sense in either it's new or original form.

What does calling `PublisherBuilder.toString()` or `Publisher.toString()` actually test?

Reply:
I guess it's a hidden `null` check. We should pass the object directly and implement a null check properly.